### PR TITLE
snapshot status

### PR DESCRIPTION
**Pull request by xiang90**
Snapshot status returns status of a database snapshot file

```
+----------+----------+------------+------------+
|   HASH   | REVISION | TOTAL KEYS | TOTAL SIZE |
+----------+----------+------------+------------+
| 23674cdd |    85716 |      85717 | 5.5 MB     |
+----------+----------+------------+------------+
```

@colhom Higher revision means newer snapshot. This can be printed as json too for programmatically consuming.

### CODE REVIEW

File: etcdctl/ctlv3/command/snapshot_command.go [hunk 1]
```diff
@@ -266,3 +287,65 @@ func makeDB(snapdir, dbfile string) {
...
+	totalSize int64
+}
+
+func dbStatus(p string) dbstatus {
+	ds := dbstatus{}
+
+	db, err := bolt.Open(p, 0600, nil)
+	if err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		os.Exit(1)
```
Comment (heyitsanthony) [inline etcdctl/ctlv3/command/snapshot_command.go]: `ExitWithError`?

Comment (xiang90) [inline etcdctl/ctlv3/command/snapshot_command.go]: right. will fix.

File: etcdctl/ctlv3/command/printer.go [hunk 2]
```diff
@@ -39,6 +40,7 @@ type printer interface {
 	MemberStatus([]statusInfo)
 
 	Alarm(v3.AlarmResponse)
+	DBStatus(dbstatus)
```
Comment (heyitsanthony) [inline etcdctl/ctlv3/command/printer.go]: not sure if this belongs in printer since it's not an rpc response

Comment (xiang90) [inline etcdctl/ctlv3/command/printer.go]: right. i will change to a simpler switch.

Comment (heyitsanthony) [inline etcdctl/ctlv3/command/printer.go]: I guess if it's doing more than just simple formatting it's OK here...

**Comment by heyitsanthony** [comment]
lgtm

File: etcdctl/ctlv3/command/printer.go (hunk 2 above)
Comment (xiang90) [inline etcdctl/ctlv3/command/printer.go]: i will think this more.

Comment (xiang90) [inline etcdctl/ctlv3/command/printer.go]: i agree with you... i guess using printer is just fine.

**Pull request merged**